sdk/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -241,6 +246,35 @@ public AggregationKind getKind() {
    */
   public abstract T getAggregate();
 
+  /** Returns the naming context associated with this counter. */
+  public NameContext getNameContext() {
+    return this.name.getContext();
+  }
+
+  /** Sets the naming context associated with this counter. */
+  public void setNameContext(NameContext context, NameType type) {
+    if (context != null) {
+      switch (type) {
+          case UNSTRUCTURED:
+            throw new IllegalArgumentException("Unstructured counters have no context");
+          case ORIGINAL:
+            this.name = new StructuredNameOriginal(this.name.getName(), context);
```

Comment:
Should it instead throw an exception if we get no context along with type ORIGINAL/OPTIMIZED? Doesn't that represent a bug?

What if this instead just tried to create the StructuredNameOriginal and that had a Preconditions.checkNotNull that imposed the appropriate requirements?